Source/Core/VideoCommon/UberShaderPixel.cpp

```diff
@@ -188,32 +188,43 @@ ShaderCode GenPixelShader(APIType ApiType, const ShaderHostConfig& host_config,
   // =====================
   //   Texture Sampling
   // =====================
+  
+  auto write_return = [&](const char* idx) {
```

Comment:
Given there's only about two things that need to be captured, making them explicit in the capture itself couldn't hurt. 

Also considering all the string inputs are quite small, they'd likely be candidates for SSO. You may want to consider a `const std::string&` here, which would let you use `std::to_string()` instead of a 100 char buffer and `snprintf`.

Reply:
No problems with that, that's actually what I do normally. I didn't want to use `std::string` in an emulator though because it's _some_ overhead.

Reply:
> because it's *some* overhead

This is likely negligible overhead. 16 char (or 22/23 in libc++'s case, iirc) buffers on the stack with no heap allocation is still more space efficient than a 100 char buffer on the stack.

Reply:
That code is called once during cache generation so I totally agree, I was just trying to be respectful and guess my _audience_ :P I thought that emulator developers would prefer being on the safe side.
I've changed it to `std::string` and it's much cleaner.

Btw, in C++17 there is `string_view` which has the best of both worlds, as it can wrap  both `const char*` (constexpr) and `string` (runtime), but has the same interface as `string`. That's actually the guideline on the projects I work on.
Sometimes, creating `std::string`s just to pass them to functions that take `const std::string&`s actually has a measurable overhead. But all of this is definitely off topic for this tiny change :P

Reply:
Oh now I remember why I choose `const char*` - `PixelShaderGen` is the first class I started working on and that uses `const char*` everywhere. I'm fine with whatever, but just make sure someone else won't pass by and ask me why I used `string` :P

Reply:
> Btw, in C++17 there is `string_view` ...

Yep but (unfortunately), I don't think we target compiler version that support it across the board (personally, I'd bump our compiler requirements for it in a heartbeat as well as for other stuff, if I had the choice that is).